### PR TITLE
[builder] Refactor project builder into distinct functions for each component

### PR DESCRIPTION
**Pull request by jaylees14**
It felt as though having one function in `ProjectBuilder` was getting quite messy, so I've separated it out into smaller functions. Some could do with a tidy up themselves, but I thought this was a good first step

### CODE REVIEW

**Review by willjvsmith — APPROVED** [review]
Look very sensible to me 🚀

**Review by Lewiky — APPROVED** [review]
Yeah I like this

File: src/main/scala/temple/builder/project/ProjectConfig.scala [hunk 1]
```diff
@@ -9,12 +10,13 @@ object ProjectConfig {
...
-  val defaultLanguage: ServiceLanguage = ServiceLanguage.Go
-  val defaultDatabase: Database        = Database.Postgres
-  val defaultAuth: ServiceAuth         = ServiceAuth.Email
-  val authPort: Int                    = 1024
-  val authMetricPort: Int              = 1025
-  val serviceStartPort: Int            = 1026
+  val defaultLanguage: ServiceLanguage  = ServiceLanguage.Go
+  val defaultDatabase: Database         = Database.Postgres
+  val defaultAuth: ServiceAuth          = ServiceAuth.Email
+  val defaultPreparedType: PreparedType = PreparedType.DollarNumbers
```
Comment (c-harding) [inline src/main/scala/temple/builder/project/ProjectConfig.scala]: Shouldn’t this be a function of the language?

**Review by c-harding — APPROVED** [review]
lgtm, just one question

File: src/main/scala/temple/builder/project/ProjectConfig.scala (hunk 1 above)
Comment (jaylees14) [reply src/main/scala/temple/builder/project/ProjectConfig.scala]: Good point, I'll fix

**Pull request merged**